src/Raven.Client/Documents/Session/AbstractDocumentQuery.cs

```diff
@@ -224,6 +225,16 @@ public void WaitForNonStaleResults(TimeSpan? waitTimeout = null)
             Timeout = waitTimeout ?? DefaultTimeout;
         }
 
+        internal LazyQueryOperation<T> GetLazyQueryOperation()
+        {
+            var beforeQueryExecutedEventArgs = new BeforeQueryEventArgs(TheSession, this);
```

Comment:
Shouldn't this only be called when QueryOperation is `null`? Check `InitSync` and `InitAsync` in DocumentQuery.